aptos-move/framework/aptos-stdlib/sources/data_structures/ordered_map.move

```diff
@@ -0,0 +1,529 @@
+                key: _,
+                value: old_value,
+            } = self.entries.replace(index, Entry { key, value });
+            option::some(old_value)
+        } else {
+            self.entries.insert(index, Entry { key, value });
+            option::none()
+        }
+    }
+
+    /// Remove a key/value pair from the map.
+    /// Aborts with EKEY_NOT_FOUND if `key` doesn't exist.
+    public fun remove<K: drop, V>(self: &mut OrderedMap<K, V>, key: &K): V {
+        let len = self.entries.length();
+        let index = binary_search(key, &self.entries, 0, len);
+        assert!(index < len, error::invalid_argument(EKEY_NOT_FOUND));
+        let Entry { key: old_key, value } = self.entries.remove(index);
+        assert!(key == &old_key, error::invalid_argument(EKEY_NOT_FOUND));
+        value
+    }
+
+    /// Returns whether map contains a given key.
+    public fun contains<K, V>(self: &OrderedMap<K, V>, key: &K): bool {
+        !self.find(key).iter_is_end()
+    }
+
+    /// Changes the key, with keeping the same value attached to it
+    /// Aborts with EKEY_NOT_FOUND if `old_key` doesn't exist.
+    /// Aborts with ENEW_KEY_NOT_IN_ORDER if `new_key` doesn't keep the order `old_key` was in.
+    public(friend) fun replace_key_inplace<K: drop, V>(self: &mut OrderedMap<K, V>, old_key: &K, new_key: K) {
+        let len = self.entries.length();
+        let index = binary_search(old_key, &self.entries, 0, len);
+        assert!(index < len, error::invalid_argument(EKEY_NOT_FOUND));
+
+        if (index > 0) {
+            assert!(cmp::compare(&self.entries[index - 1].key, &new_key).is_less_than(), error::invalid_argument(ENEW_KEY_NOT_IN_ORDER))
+        };
+
+        if (index + 1 < len) {
+            assert!(cmp::compare(&new_key, &self.entries[index + 1].key).is_less_than(), error::invalid_argument(ENEW_KEY_NOT_IN_ORDER))
+        };
+
+        let entry = self.entries.borrow_mut(index);
+        assert!(old_key == &entry.key, error::invalid_argument(EKEY_NOT_FOUND));
+        entry.key = new_key;
+    }
+
+    /// Takes all elements from `other` and adds them to `self`.
+    public fun append<K, V>(self: &mut OrderedMap<K, V>, other: OrderedMap<K, V>) {
+        let OrderedMap::SortedVectorMap {
+            entries: other_entries,
+        } = other;
+
+        if (other_entries.is_empty()) {
+            other_entries.destroy_empty();
+            return;
+        };
+
+        if (self.entries.is_empty()) {
+            self.entries.append(other_entries);
+            return;
+        };
+
+        // Optimization: if all elements in `other` are larger than all elements in `self`, we can just move them over.
+        if (cmp::compare(&self.entries.borrow(self.entries.length() - 1).key, &other_entries.borrow(0).key).is_less_than()) {
+            self.entries.append(other_entries);
+            return;
+        };
+
+        // TODO: can be implemented more efficiently, as we know both maps are already sorted.
```

Comment:
https://leetcode.com/problems/merge-two-sorted-lists/

Reply:
yeah, but issue is here we have vectors, not lists, and so how to do it in-place, without moving too many elements.

Easiest is to can either create a new vector, populate, and then swap (I'll probably just do that)

if "vector" here had capacity, I could start from the end, without temporary storage, but I don't have a way to create empty values here (i.e. have this work without value being drop+copy)